app/models/spree/user_decorator.rb

```diff
@@ -4,7 +4,7 @@
   devise :omniauthable
 
   def apply_omniauth(omniauth)
-    if %w(facebook google_oauth2).include? omniauth['provider']
+    if omniauth.fetch('info', {})['email'].present?
```

Comment:
Can we use `.dig`? I don't remember off the top of my head what version of Ruby added that.